apollo-router/src/query_planner/plan.rs

```diff
@@ -382,6 +383,60 @@ impl PlanNode {
             }
         }
     }
+
+    pub(crate) fn parse_subgraph_operations(
+        &mut self,
+        schemas: &HashMap<String, Arc<Valid<Schema>>>,
+    ) {
+        match self {
+            // Traverse the tree to get down to the leaf nodes
+            PlanNode::Condition {
+                condition: _,
+                if_clause,
+                else_clause,
+            } => {
+                if let Some(node) = if_clause {
+                    node.parse_subgraph_operations(schemas);
+                }
+                if let Some(node) = else_clause {
+                    node.parse_subgraph_operations(schemas);
+                }
+            }
+            PlanNode::Defer { primary, deferred } => {
+                if let Some(node) = primary.node.as_mut() {
+                    node.parse_subgraph_operations(schemas);
+                }
+                for deferred_node in deferred {
+                    if let Some(node) = deferred_node.node.take() {
+                        let mut new_node = (*node).clone();
+                        new_node.parse_subgraph_operations(schemas);
+                        deferred_node.node = Some(Arc::new(new_node));
+                    }
```

Comment:
This function will go away entirely with my lazy-initialization proposal above, but this pattern can be improved with [`Arc::make_mut`](https://doc.rust-lang.org/std/sync/struct.Arc.html#method.make_mut) which only clones if the reference count is above 1.
```suggestion
                    if let Some(node) = &mut deferred_node.node {
                        Arc::make_mut(node).parse_subgraph_operations(schemas);
                    }
```